app/models/project/phase.rb

```diff
@@ -84,7 +86,14 @@ def range_incomplete?
   end
 
   def validate_date_range
-    errors.add(:date_range, :start_date_must_be_before_finish_date) if range_set? && (start_date > finish_date)
+    if range_set? && (start_date > finish_date)
+      date_field, error_message = if finish_date_changed?
+                                    %i[finish_date must_be_after_start_date]
+                                  else
+                                    %i[start_date must_be_before_finish_date]
+                                  end
+      errors.add(date_field, error_message)
```

Comment:
I think it is simpler to do more directly:
```suggestion
      if finish_date_changed?
        errors.add(:finish_date, :must_be_after_start_date)
      else
        errors.add(:start_date, :must_be_before_finish_date)
      end
```